### PR TITLE
Extend AntreaConfig to Support Antrea-NSX Registration Workflow

### PR DESCRIPTION
**Pull request by liu4480**
### What this PR does / why we need it
This patch extends antreaConfig to support antreaNsx, once antreaNsx is enabled in antreaConfig, addon controller will
1. create nsxServiceAccount
2. create providerServiceAccount
3. update values.yaml in xxxx-antrea-package secret  with antreaNsx enabled

upon the workload cluster is deleted, addon controller will
1. delete nsxServiceAccount
2. delete providerServiceAccount
3. delete xxxx-antrea-package secret

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
1. create a cluster in namespace antrea-test
2. edit antreaConfig in antrea-test, add 
```
  antreaNsx:
    enable: true
```
3. check if nsxServiceAccount and providerServiceAccount are created in antrea-test
4. check if there is antreaNsx enabled in secret whose name contains antrea-package in its data, use based64 -d to decode the content
5. delete the cluster, check if nsxServiceAccount and providerServiceAccount are deleted
6. check if secret in step 4 is deleted
```
  antreaNsx:
    enable: true
```

### Release note

```release-note

```

### CODE REVIEW

**Comment by liu4480** [comment]
as https://github.com/vmware-tanzu/tanzu-framework/pull/3816 is based on a forked repo, propose a new pr to replace 3816

**Comment by alfredthenarwhal** [comment]
> @navidshaikh:  /test install-vc7
Commit: ea7b3ac

Tests failed! Build no: 3716

**Comment by alfredthenarwhal** [comment]
> @liu4480:  /test install-vc7
Commit: 9105d2c

Build failed! Build no: 3740

**Comment by alfredthenarwhal** [comment]
> @liu4480:  /test install-vc7
Commit: 9105d2c

Tests failed! Build no: 3753

**Comment by knabben** [comment]
reviewed the code base in the old PR

LGTM

**Review by knabben — APPROVED** [review]

**Comment by alfredthenarwhal** [comment]
> @navidshaikh:  /test install-vc7
Commit: 0cd5815

Tests failed! Build no: 3968

**Review by jeffwubj — APPROVED** [review]

**Comment by liu4480** [comment]
can you help to choose squash and merge when merging the pr, thanks

**Comment by liu4480** [comment]
this pr depends on another patch, let's wait for the patch to be ready. CC @edwardbadboy

**Comment by alfredthenarwhal** [comment]
> @liu4480:  /test install-vc7
Commit: c88a895

Tests failed! Build no: 4080

**Comment by alfredthenarwhal** [comment]
> @liu4480:  /test install-vc7
Commit: c88a895

Tests failed! Build no: 4081

File: addons/controllers/antrea/antreaconfig_controller.go [hunk 1]
```diff
@@ -26,16 +29,35 @@ import (
...
+	nsxoperatorapi "github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
 	addontypes "github.com/vmware-tanzu/tanzu-framework/addons/pkg/types"
 
 	cutil "github.com/vmware-tanzu/tanzu-framework/addons/controllers/utils"
 	addonconfig "github.com/vmware-tanzu/tanzu-framework/addons/pkg/config"
 	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
 	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/util"
 	"github.com/vmware-tanzu/tanzu-framework/addons/predicates"
-	cniv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/addonconfigs/cni/v1alpha1"
+	cniv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/addonconfigs/cni/v1alpha2"
```
Comment (edwardbadboy) [inline addons/controllers/antrea/antreaconfig_controller.go:40]: Should we change this `cnialpha1` to `cniv1alpha2`? Same comment for `cni/v1alpha2` import in file addons/controllers/antrea/antreaconfig_util.go.

Comment (liu4480) [reply addons/controllers/antrea/antreaconfig_controller.go]: thanks, done

**Comment by alfredthenarwhal** [comment]
> @liu4480:  /test install-vc7
Commit: e44fff7

Tests failed! Build no: 4251

**Comment by alfredthenarwhal** [comment]
> @liu4480:  /test install-vc7
Commit: 5ffad23

Tests failed! Build no: 4260

**Comment by alfredthenarwhal** [comment]
> @liu4480:  /test install-vc7
Commit: 5ffad23

Tests failed! Build no: 4261

**Review by wjun — APPROVED** [review]

**Pull request merged**